File: loop_transformations/sources/split.1.c
/*
* @@name:       split.1
* @@type:       C   
* @@operation:  run
* @@expect:     success
* @@version:    omp_6.0
*/
void adder(int m, int n, float* A, float* B, float* C){

  // Split out vectorizable portion of loop.
  #pragma omp split counts(m,1,omp_fill)
  for(int i = 0; i<n; i++){
    A[i] = A[m] + B[i] + C[i];
  }
}

void adder_equivalent(int m, int n, float* A, float* B, float* C){

  for(int i = 0; i<m; i++){
    A[i] = A[m] + B[i] + C[i];
  }

  for(int i = m; i<m+1; i++){
    A[i] = A[m] + B[i] + C[i];
  }

  for(int i = m+1; i<n; i++){
    A[i] = A[m] + B[i] + C[i];
  }

}

File: loop_transformations/sources/split.1.f90
! @@name:       split.1
! @@type:       F-free
! @@operation:  compile
! @@expect:     success
! @@version:    omp_6.0
subroutine adder(m, n, A, B, C)
    implicit none
    integer, intent(in)    :: m,n
    real,    intent(inout) :: A(0:n-1)
    real,    intent(in)    :: B(0:n-1),C(0:n-1)
    integer                :: i

    !$omp split counts(m,1,omp_fill)
    do i = 0,n-1
      A(i) = A(m) + B(i) + C(i)
    end do
end subroutine

subroutine adder_equivalent(m, n, A, B, C)
    implicit none
    integer, intent(in)    :: m,n
    real,    intent(inout) :: A(0:n-1)
    real,    intent(in)    :: B(0:n-1),C(0:n-1)
    integer                :: i

    do i = 0,m-1
      A(i) = A(m) + B(i) + C(i)
    end do

    do i = m,m
      A(i) = A(m) + B(i) + C(i)
    end do

    do i = m+1,n-1
      A(i) = A(m) + B(i) + C(i)
    end do

end subroutine
